omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,300 @@
+			IdentifierSource poolSource = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid);
+			
+			if (!StringUtils.isBlank(batchSize)) {
+				if(StringUtils.isNumeric(batchSize)) {
+					identifierSource.setBatchSize(Integer.parseInt(batchSize));
+				}else {
+		        		exception.add("batch size");
+		        }
+	        }
+			if (!StringUtils.isBlank(minPoolSize)) {
+				if(StringUtils.isNumeric(minPoolSize)) {
+					identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize));
+					Set<PooledIdentifier> pooledIdentifiers = new LinkedHashSet<PooledIdentifier>();
+					for(Integer counter = 0; counter < Integer.parseInt(minPoolSize); counter++) {
+						pooledIdentifiers.add(new PooledIdentifier(identifierSource, "ABC0" + counter));
+					}
+					identifierSource.setIdentifiers(pooledIdentifiers);
+				}else {
+		        		exception.add("min pool size");
+		        }
+	        }
+			if (!StringUtils.isBlank(sequential)) {
+				if(sequential.toLowerCase().equals("true") || sequential.toLowerCase().equals("false")) {
+					identifierSource.setSequential(Boolean.parseBoolean(sequential));
+				}else {
+		        		exception.add("sequential");
+		        }
+	        }
+			if (!StringUtils.isBlank(refillWithScheduledTask)) {
+				if(refillWithScheduledTask.toLowerCase().equals("true") || refillWithScheduledTask.toLowerCase().equals("false")) {
+					identifierSource.setSequential(Boolean.parseBoolean(refillWithScheduledTask));
+				}else {
+		        		exception.add("refill");
+		        }
+	        }
+			if(exception.toString().length() > 2) {
+				throw new ConversionException("The values of the following inputs are missing or invalid: " + exception.toString());
+			}
+			
+			identifierSource.setIdentifierType(patientIdentifierType);
+			identifierSource.setName(name);
+			identifierSource.setDescription(description);
+			identifierSource.setSource(poolSource);
+			
+			savedIdentifierSource =  save(identifierSource);
+			
+		}
+		
+		return savedIdentifierSource;
+    }
+	
+	@Override
+	public IdentifierSource save(IdentifierSource identifierSource) {
+		return Context.getService(IdentifierSourceService.class).saveIdentifierSource(identifierSource);
+	}
+	
+	@Override
+	public IdentifierSource getByUniqueId(String uniqueId) {
+		return Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uniqueId);
+	}
+	
+	@Override
+    protected void delete(IdentifierSource identifierSource, String reason, RequestContext context) throws ResponseException {
+		Context.getService(IdentifierSourceService.class).retireIdentifierSource(identifierSource, reason);
+    }
+	
+	@Override
+	public void purge(IdentifierSource identifierSource, RequestContext context) throws ResponseException {
+		Context.getService(IdentifierSourceService.class).purgeIdentifierSource(identifierSource);
+	}
+	
+	private String checkInput(Object input) {
+		if (input == null | input.toString().length() == 0) {
```

Comment:
Did you mean to use "I" or "II"?

Reply:
What is the meaning of checkInput?

Reply:
changed it to `validateInput()` and used `||` for or

Reply:
removed the method()